snr.py

```diff
@@ -0,0 +1,287 @@
+        Electrons per pixel from the read noise with units of electrons/pixel.
+        Default is 0 * astropy.units.electron / astropy.units.pixel.
+    gain : `~astropy.units.Quantity`, optional
+        Gain of the CCD with units of electrons/ADU. Default is
+        1 * astropy.units.electron / astropy.units.adu such that the
+        contribution to the error due to the gain is assumed to be small.
+    ad_err : `~astropy.units.Quantity`, optional
+        An estimate of the 1 sigma error within the A/D converter with units of
+        adu/pixel. Default is set to
+        sqrt(0.289) * astropy.units.adu / astropy.units.pixel
+        (Merline & Howell, 1995).
+
+    Returns
+    -------
+    t : `~astropy.units.Quantity`
+        The exposure time needed (in seconds) to achieve the given signal
+        to noise ratio.
+    """
+    readnoise = _get_shotnoise(readnoise)
+    gain_err = _get_shotnoise(gain * ad_err)
+
+    # solve t with the quadratic equation (pg. 57 of Howell 2000)
+    A = countrate ** 2
+    B = (-1) * snr ** 2 * (countrate + npix * (background_rate +
+                                               darkcurrent_rate))
+    C = (-1) * snr ** 2 * npix * readnoise ** 2
+
+    t = (-B + np.sqrt(B ** 2 - 4 * A * C)) / (2 * A)
+
+    if gain_err.value > 1 or not np.isfinite(n_background.value):
+        # solve t numerically
+        t = fsolve(_t_with_small_errs, t, args=(background_rate,
+                                                darkcurrent_rate,
+                                                gain_err, readnoise, countrate,
+                                                npix, n_background))
+        if not hasattr(t, 'unit'):
+            t = t * u.s
+
+    return t
+
+
+def test_howell_snr_calc():
+    """
+    A test to check that the math in howell_snr() is done correctly.
+    Based on the worked example in "A Handbook to CCD Astronomy",
+    Steven Howell, 2000, pg. 56-57
+    """
+    t = 300 * u.s
+    readnoise = 5 * u.electron / u.pixel
+    darkcurrent = 22 * t * u.electron / u.pixel / u.hr
+    gain = 5 * u.electron / u.adu
+    background = 620 * u.adu / u.pixel * gain
+    n_background = 200 * u.pixel
+    npix = 1 * u.pixel
+    counts = 24013 * u.adu * gain
+
+    result = howell_snr(counts, npix=npix, background=background,
+                        darkcurrent=darkcurrent, readnoise=readnoise,
+                        gain=gain, n_background=n_background)
+
+    # the value of the answer given in the text
+    answer = 342 * np.sqrt(1 * u.electron)
+
+    # allow error to be +/- 1 for this test
+    assert_quantity_allclose(result, answer, atol=1 * np.sqrt(1 * u.electron))
+
+
+def test_snr_bright_object():
+    """
+    Test that howell_snr() returns sqrt(counts), the expected value
+    for a bright target.
+    """
+    counts = 25e5 * u.electron
+    result = howell_snr(counts)
+    answer = np.sqrt(counts)
+
+    assert_quantity_allclose(result, answer, rtol=1e-7)
```

Comment:
Strictly speaking, you don't need to provide `rtol` here because 1e-7 is the default anyway.
```suggestion
    assert_quantity_allclose(result, answer)
```